ENIGMAsystem/SHELL/Graphics_Systems/General/GSmodel.cpp

```diff
@@ -212,11 +214,18 @@ void d3d_model_draw(int id, gs_scalar x, gs_scalar y, gs_scalar z, int texId) {
 
 void d3d_model_primitive_begin(int id, int kind, int format) {
   enigma::Model* model = enigma::models[id];
+  #ifdef DEBUG_MODE
+  if (model->current_primitive) {
+    show_error("d3d_model_primitive_begin called before d3d_model_primitive_end " \
```

Comment:
This stops being an error, that way.

Reply:
I tested in GM8.1 and GMSv1.4 and I guess this was never actually an error message in run or debug mode either. So now I am not sure if we want to do this.
```gml
mymodel = d3d_model_create();
d3d_model_primitive_begin(mymodel, pr_trianglelist);
d3d_model_primitive_begin(mymodel, pr_trianglelist);
d3d_model_primitive_end(mymodel);
```

Reply:
Alright, I tweaked it:
![Primitive Begin Called Twice](https://user-images.githubusercontent.com/3212801/44559370-a0489d80-a717-11e8-8570-439771708db3.png)

Reply:
If you stop using `new` every time the user begins a primitive, this will stop being an error.